Can't compile due to trailing backslash in gcc args on Windows

Version: 1.0-git:e6575c2b

On Windows, the compiler adds trailing backslash in include and link path when calling gcc, which seems to invalidate these paths (at least for the gcc included in MSYS2 and Haskell Platform). Could you change the behavior?

e.g. the call to gcc below works iff the trailing `\\` are removed in `-I` and `-L ...\\idris-1.0\\rts\\`.
```
idris ffi.idr -o ffi.exe
C:\Users\w\AppData\Local\Temp\idris9961491.c:3:23: fatal error: idris_rts.h: No such file or directory
compilation terminated.
FAILURE: "gcc" ["-O2","-fwrapv","-fno-strict-overflow","-DHAS_PTHREAD","-DIDRIS_ENABLE_STATS","-I.",".\\lib.o","C:\\Users\\w\\AppData\\Local\\Temp
\\idris9961491.c","-LC:\\Users\\w\\AppData\\Roaming\\cabal\\x86_64-windows-ghc-8.0.2\\idris-1.0\\rts\\","-lidris_rts","-lpthread","-IC:\\Users\\w\
\AppData\\Roaming\\cabal\\x86_64-windows-ghc-8.0.2\\idris-1.0\\rts\\","-lm","-I.","-I.","-IC:\\Users\\w\\AppData\\Roaming\\cabal\\x86_64-windows-g
hc-8.0.2\\idris-1.0\\libs\\prelude","-IC:\\Users\\w\\AppData\\Roaming\\cabal\\x86_64-windows-ghc-8.0.2\\idris-1.0\\libs\\base","-Wl,--stack,167772
16","-o","ffi.exe"]
```

--- a/src/IRTS/System.hs
+++ b/src/IRTS/System.hs
@@ -30,7 +30,8 @@ import Control.Applicative ((<$>))
 import Data.List.Split
 import Data.Maybe (fromMaybe)
 import System.Environment
-import System.FilePath (addTrailingPathSeparator, (</>))
+import System.FilePath (addTrailingPathSeparator, dropTrailingPathSeparator,
+                        (</>))
 
 
 getIdrisDataDir :: IO String
@@ -81,7 +82,7 @@ gmpLib = []
 #endif
 
 getLibFlags = do dir <- getIdrisCRTSDir
-                 return $ ["-L" ++ dir,
+                 return $ ["-L" ++ dropTrailingPathSeparator dir,
                            "-lidris_rts"] ++ extraLib ++ gmpLib ++ ["-lpthread"]
 
 getIdrisLibDir = addTrailingPathSeparator <$> overrideIdrisSubDirWith "libs" "IDRIS_LIBRARY_PATH"
@@ -97,4 +98,4 @@ getIdrisCRTSDir = do
   return $ addTrailingPathSeparator (ddir </> "rts")
 
 getIncFlags = do dir <- getIdrisCRTSDir
-                 return $ ("-I" ++ dir) : extraInclude
+                 return $ ("-I" ++ dropTrailingPathSeparator dir) : extraInclude

